UOP1_Project/Assets/Scripts/Pool/IPool.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace OP1.Pool
+{
+	/// <summary>
+	/// Represents a collection that pools objects of T.
+	/// </summary>
+	/// <typeparam name="T">Specifies the type of elements in the pool.</typeparam>
+	public interface IPool<T>
+	{
```

Comment:
While this PR _is_ using Factories in every Pool implementation, it is not necessary. Unlike Request and Return functionality, a pool could still pool without a factory. I think it would be a worse pool, but a pool nonetheless. So I'd prefer not to enforce factory usage from the interface.

Reply:
I'd argue that that would simply be a Stack with different method names, but I guess you're right, there's no need to couple them together.

Reply:
It doesn't even have to be a Stack. I just used the collection I thought was best for the job. Another Pool implementation might want individual index access and use a List. But that wouldn't be possible if I said all pools must use a Stack.